api/adaptive_load/adaptive_load.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/benchmark_result.proto";
+import "api/adaptive_load/metrics_plugin.proto";
+import "api/adaptive_load/metric_spec.proto";
+import "api/adaptive_load/step_controller.proto";
+import "api/client/options.proto";
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+import "validate/validate.proto";
+
+// Parameters describing the adjusting and testing stages of an adaptive load
+// session, which consists of a series of Nighthawk benchmarks probing for
+// the optimal load on the system, followed by a longer benchmark to validate
+// the values. Load adjustments are calculated by the selected StepController
+// plugin. Metrics can come from Nighthawk stats and counters or
+// platform-specific data sources via MetricsPlugins.
+message AdaptiveLoadSessionSpec {
+  // Settings for MetricsPlugins that obtain metrics from outside sources.
+  // An entry is required for every plugin referred to by metric_thresholds,
+  // other than the "builtin" plugin.
+  repeated MetricsPluginConfig metrics_plugin_configs = 1;
+  // Metrics and thresholds that determine load adjustments.
+  repeated MetricSpecWithThreshold metric_thresholds = 2 [(validate.rules).repeated .min_items = 1];
+  // Metrics that are collected and included in the output but not taken into
+  // account when adjusting the load.
+  repeated MetricSpec informational_metric_specs = 3;
```

Comment:
What is this going to be used for in practice?

Reply:
This could be useful for both debugging and visualization and comes at almost no added cost. Originally I was going to have comments pointing out that defining a metric with a dummy threshold and weight 0 could be used for this purpose, but then realized it could be explicit.

Reply:
The builtin metrics I'm planning could be interesting to track as informational even if they are not used to control the RPS:
- attempted RPS
- achieved RPS
- success ratio (2xx responses / total responses)
- send rate (how much of the attempted RPS Nighthawk actually managed to send)